### PR TITLE
Initial code import

### PR DESCRIPTION
**Pull request by rbradford**
- acpi_tables: Add initial ACPI tables support
- acpi_tables: aml: Add support for generating AML name paths
- acpi_tables: aml: Implement numbers
- acpi_tables: aml: Add support for named definitions
- acpi_tables: aml: Add Package support
- acpi_tables: aml: Generate EISA name IDs
- acpi_tables: aml: Add resource templates and Memory32Fixed
- acpi_tables: aml: Add support for address spaces
- acpi_tables: aml: Add support for creating IO and interrupt resources
- acpi_tables: aml: Add support for device and scope
- acpi_tables: aml: Implement AML Zero, One, Ones
- acpi_tables: sdt: Add ability to add to the table from a slice
- acpi_tables: aml: Rename to_bytes() to to_aml_bytes()
- acpi_tables: aml: Add support for strings
- acpi_tables: aml: Add support for Method and Return
- acpi_tables: Fix rustfmt failure by removing extra ";"
- acpi_tables: Add PkgLength variant that does not include itself
- acpi_tables: aml: Add support for field definitions
- acpi_tables: aml: Add support for OpRegion
- acpi_tables: aml: Add "if" with local variables and arguments
- acpi_tables: aml: Add support for mutexes
- acpi_tables: aml: Add support for device notification
- acpi_tables: aml: Add support for while loops
- acpi_tables: aml: Add support for calling methods
- acpi_tables: Add support for Buffer objects
- acpi_tables: aml: Add support for binary operators
- acpi_tables: aml: Add support for creating fields
- build: Bump linux-loader and vm-memory dependencies
- build: Bump vm-memory and linux-loader dependencies
- build(deps): bump vm-memory from 0.2.0 to 0.2.1
- vmm: acpi_tables: Fix PM timer I/O port width
- build: Fix vm-memory bump build error
- cargo: Move to crates.io vm-memory 0.4.0
- build(deps): bump vm-memory from 0.4.0 to 0.5.0
- acpi_tables: Address Rust 1.51.0 clippy issue (upper_case_acronyms)
- acpi_tables: Address Rust 1.51.0 clippy issue (vec_init_then_push)
- aml: make constructor field order consistent
- acpi: Implement SPCR on AArch64
- vmm, acpi: Add DSM method to ACPI
- clippy: Address the issue 'needless-borrow'
- build: Move from patched vm-memory version to released version
- build: bump vm-memory from 0.6.0 to 0.7.0
- Revert "build: bump vm-memory from 0.6.0 to 0.7.0"
- acpi_tables: aml: Add Aml trait implementation for vector appending
- acpi_tables: aml: Implement Aml::append_aml_bytes() for constants
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Path
- acpi_tables: aml: Implement Aml::append_aml_bytes() for number types
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Name
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Package
- acpi_tables: aml: Avoid extra vector allocation in number types
- acpi_tables: aml: Implement Aml::append_aml_bytes() for EisaName
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Usize
- acpi_tables: aml: Implement Aml::append_aml_bytes() for AmlStr and AmlString
- acpi_tables: aml: Implement Aml::append_aml_bytes() for ResourceTemplate
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Memory32Fixed
- acpi_tables: aml: Avoid allocating temporary vector in Memory32Fixed
- acpi_tables: aml: Avoid allocating temporary vector in AddressSpace
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Io
- acpi_tables: aml: Avoid allocating temporary vector in Io
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Interrupt
- acpi_tables: aml: Avoid allocating temporary vector in Interrupt
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Device
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Scope
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Method
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Return
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Field
- acpi_tables: aml: Implement Aml::append_aml_bytes() for OpRegion
- acpi_tables: aml: Implement Aml::append_aml_bytes() for If
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Equal
- acpi_tables: aml: Implement Aml::append_aml_bytes() for LessThan
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Arg
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Local
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Store
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Mutex
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Acquire
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Release
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Notify
- acpi_tables: aml: Implement Aml::append_aml_bytes() for While
- acpi_tables: aml: Implement Aml::append_aml_bytes() for !binary_op macro
- acpi_tables: aml: Implement Aml::append_aml_bytes() for MethodCall
- acpi_tables: aml: Implement Aml::append_aml_bytes() for Buffer
- acpi_tables: aml: Implement Aml::append_aml_bytes() for CreateField
- acpi_tables: aml: Use Aml::append_aml_bytes()
- acpi_tables: aml: Replace use of Vec::append with Vec::extend_from_slice
- acpi_tables: add safety comment for `impl ByteValued`
- deps: Bump vm-memory to 0.7.0
- acpi_tables: Fix clippy (unnecessary_to_owned) issues
- build: migrate to Rust 2021 edition
- build: bump virtio-queue from 0.2.0 to 0.3.0
- build: Fix beta clippy issue (unnecessary_to_owned)
- vmm: acpi: Use ACPI platform device addresses from DeviceManager
- build: Bulk dependencies update
- acpi_tables: add a safety comment for the write function
- build: Bulk update dependencies
- acpi_tables: sdt: Implement Std::is_empty()
- README: Replace template README
- misc: Remove LICENSE-BSD-3-Clause
- misc: Update CODEOWNERS
- build: Add GitHub actions

### CODE REVIEW

**Review by aghecenco — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1 @@
-# Add the list of code owners here (using their GitHub username)
-*
+* @jiangliu @liuw @rbradford @sboeuf
```
[the remark this answers is not in the document]
Comment (rbradford) [reply CODEOWNERS:1]: @liuw apparently you don't exist?

@liuw @jiangliu @sboeuf just checking you're okay with being reviewers here? You have either contributed to this code or have expressed interest in it.

[the remark this answers is not in the document]
Comment (liuw) [reply CODEOWNERS:1]: I'm fine with being a reviewer.

> @liuw apparently you don't exist?

Perhaps I'm not a member of the correct organization?

[the remark this answers is not in the document]
Comment (rbradford) [reply CODEOWNERS:1]: @liuw you are listed as a member of the rust-vmm organisation.

**Review by sboeuf — APPROVED** [review]

**Comment by rbradford** [comment]
@aghenco Unfortunately you merged this with a merge commit.

**Pull request merged**